controller/cmd/identity/main.go

```diff
@@ -102,27 +109,70 @@ func Main(args []string) {
 		}
 	}
 
-	ca := tls.NewCA(*creds, validity)
+	expectedName := fmt.Sprintf("identity.%s.%s", controllerNS, trustDomain)
+	issuerEvent := make(chan struct{})
+	issuerError := make(chan error)
+
+	//
+	// Create and start FS creds watcher
+	//
+	watcher := idctl.NewFsCredsWatcher(*issuerPath, issuerEvent, issuerError)
+	go func() {
+		if err := watcher.StartWatching(ctx); err != nil {
+			log.Fatalf("Failed to start creds watcher: %s", err)
+		}
+	}()
 
-	k8s, err := k8s.NewAPI(*kubeConfigPath, "", "", 0)
+	//
+	// Create k8s API
+	//
+	k8sAPI, err := k8s.NewAPI(*kubeConfigPath, "", "", 0)
 	if err != nil {
 		log.Fatalf("Failed to load kubeconfig: %s: %s", *kubeConfigPath, err)
 	}
-	v, err := idctl.NewK8sTokenValidator(k8s, dom)
+	v, err := idctl.NewK8sTokenValidator(k8sAPI, dom)
 	if err != nil {
 		log.Fatalf("Failed to initialize identity service: %s", err)
 	}
 
-	svc := identity.NewService(v, ca)
+	// Create K8s event recorder
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{
+		Interface: k8sAPI.CoreV1().Events(controllerNS),
+	})
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: componentName})
+	deployment, err := k8sAPI.ExtensionsV1beta1().Deployments(controllerNS).Get(componentName, v1machinary.GetOptions{})
```

Comment:
it sucks that we have to lookup our deployment in order to be able to record events about it.  what a weird API 😬

Reply:
Yeah,  I was a bit perplexed by that as well... its not great, especially due to the additional permissions required...